src/common.rs

```diff
@@ -73,20 +74,42 @@ pub type BoxedFuture<T> = Box<Future<Item = T, Error = BoxedErr> + Send + Sync +
 #[derive(Debug)]
 pub struct Errors {
     /// The collection of errors.
-    pub errors: Vec<BoxedErr>,
+    errors: MsQueue<BoxedErr>,
 }
 
 impl Errors {
     /// Creates a new instance of `Errors`.
     pub fn new() -> Self {
-        Errors { errors: Vec::new() }
+        Errors {
+            errors: MsQueue::new(),
+        }
     }
 
-    /// Add an error to the error `Vec`.
-    pub fn add(&mut self, error: BoxedErr) {
+    /// Add an error to the error collection.
+    pub fn add(&self, error: BoxedErr) {
         self.errors.push(error);
     }
 
+    /// Checks if the queue contains at least one error.
+    pub fn has_error(&self) -> bool {
+        !self.errors.is_empty()
+    }
+
+    /// Removes the first error from the queue.
+    pub fn pop_err(&mut self) -> Option<BoxedErr> {
+        self.errors.try_pop()
+    }
+
+    /// Collect all errors into a `Vec`.
+    pub fn collect(&mut self) -> Vec<BoxedErr> {
+        let mut errors = Vec::new();
+        while let Some(e) = self.errors.try_pop() {
```

Comment:
Isn't there a method to pop all queue at once?

Reply:
No.

Reply:
I'll take a look if it may be added 😄